performance/performance_tuning.rst

```diff
@@ -185,6 +185,47 @@ Citus has two executor types for running SELECT queries. The desired executor ca
 
 Other than the above, there are two configuration parameters which can be useful in cases where approximations produce meaningful results. These two parameters are citus.limit_clause_row_fetch_count and citus.count_distinct_error_rate. The former sets the number of rows to fetch from each task while calculating limits while the latter sets the desired error rate when calculating approximate distinct counts. You can learn more about the applicability and usage of these parameters in the user guide sections: :ref:`count_distinct` and :ref:`limit_pushdown`.
 
+.. _subquery_perf:
+
+Subquery/CTE Network Overhead
+=============================
+
+In the best case Citus can execute queries containing subqueries and CTEs in a single step. This is usually because both the main query and subquery filter by tables' distribution column in the same way, and can be pushed down to worker nodes together. However Citus is sometimes forced to execute subqueries *before* executing the main query, copying the intermediate subquery results to other worker nodes for use by the main query. This technique is called :ref:`push_pull_execution`.
+
+It's important to be aware when subqueries are executed in a separate step, and avoid sending too much data between worker nodes. The network overhead will hurt performance.
+
+To detect if this is happening, adjust the ``max_intermediate_result_size`` limit in a new connection to the coordinator node. By default the max intermediate result size is 1GB, which is large enough to allow some inefficient queries. Try turning it down and running your queries:
```

Comment:
I had a previous comment that I'll reiterate here. Instead of detecting this via modification of this parameter, can we use EXPLAIN to show how the plan would be executed? I'd want to highlight and promote EXPLAIN as a way to diagnose/debug Citus execution and performance issues.

Reply:
Oops, I added an example of EXPLAIN in query_processing.rst and overlooked the one here.